blueflood-http/src/main/java/com/rackspacecloud/blueflood/http/QueryStringDecoderAndRouter.java

```diff
@@ -45,8 +48,30 @@ public void channelReadComplete(ChannelHandlerContext ctx) {
     }
 
     @Override
-    public void channelRead0(ChannelHandlerContext ctx, FullHttpRequest msg) throws Exception {
-        router.route(ctx, HttpRequestWithDecodedQueryParams.create(msg));
+    public void channelRead0(ChannelHandlerContext ctx, FullHttpRequest request) throws Exception {
+
+        // for POST requests, check Content-Type header
+        if ( request.getMethod() == HttpMethod.POST ) {
+            if (!mediaTypeChecker.isContentTypeValid(request.headers())) {
+                DefaultHandler.sendResponse(ctx, request,
+                        String.format("Unsupported media type for Content-Type: %s", request.headers().get(HttpHeaders.Names.CONTENT_TYPE)),
+                        HttpResponseStatus.UNSUPPORTED_MEDIA_TYPE
+                );
+                return;
+            }
+        }
+
+        // for GET or POST requests, check Accept header
+        if ( request.getMethod() == HttpMethod.GET || request.getMethod() == HttpMethod.POST ) {
+            if (!mediaTypeChecker.isAcceptValid(request.headers())) {
+                DefaultHandler.sendResponse(ctx, request,
+                        String.format("Unsupported media type for Accept: %s", request.headers().get(HttpHeaders.Names.ACCEPT)),
+                        HttpResponseStatus.UNSUPPORTED_MEDIA_TYPE
+                );
+                return;
+            }
+        }
```

Comment:
@usnavi @izrik @ChandraAddala , I moved the checked to this InboundHandler, prior to the routing of the request to the appropriate HttpRequestHandler.